lib/trino-parquet/src/main/java/io/trino/parquet/ParquetReaderUtils.java

```diff
@@ -126,4 +136,25 @@ public static byte toByteExact(int value)
         }
         return (byte) value;
     }
+
+    @SuppressWarnings("deprecation")
```

Comment:
Why? If this relies on deprecated APIs, can use non-deprecated versions?

Reply:
The annotation is carried over from existing code. It's to suppress warnings about the deprecated encodings `PLAIN_DICTIONARY` and `BIT_PACKED` which we need to handle anyway because they appear in existing parquet files.
This logic is the same as parquet-mr code `org.apache.parquet.filter2.dictionarylevel.DictionaryFilter#hasNonDictionaryPages`